tools/autograd/derivatives.yaml

```diff
@@ -378,6 +378,13 @@
   self: grad
   result: auto_linear
 
+- name: _to_copy(Tensor self, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None, bool non_blocking=False, MemoryFormat? memory_format=None) -> Tensor
+  self: grad.to(self.options(), /*non_blocking*/false, /*copy*/false)
+  result: _to_copy(self_t, dtype, layout, device, pin_memory, non_blocking, memory_format)
+  # The condition is: if dtype is not nullopt, then isDifferentiableType(*dtype)
+  # (If dtype IS nullopt, we rely on the regular check that any input requires grad).
+  output_differentiability: ["!dtype || isDifferentiableType(*dtype)"]
```

Comment:
@albanD this PR differs from the original one in that the original's output_differentiability condition was "dtype && isDifferentiableType(*dtype). That was incorrect because if `dtype` is nullopt it would set the differentiability to False. Instead, what we really wanted to do is set the differentiability to false if dtype is not nullopt and dtype is a differentiable type.

This was caught in a C++ API test, and not on the Python side, because we have some very special handling for `tensor.to('cuda:0')` in Python...

Reply:
Is there a way we can test that in python?

Reply:
I could create a new operator called `_debug_to` with the sole purpose of testing this in Python, but there is a test in C++ already so we already have test coverage for it

Reply:
Ok sounds good!